CRM/Member/Form/MembershipRenewal.php

```diff
@@ -445,7 +498,18 @@ public static function formRule($params) {
         $errors['payment_instrument_id'] = ts('Payment Method is a required field.');
       }
     }
-    return empty($errors) ? TRUE : $errors;
+
+    if (empty($errors) && !empty($params['price_set_id'])) {
```

Comment:
I'm not liking this - the 2 classes have a shared parent so let's just call a shared function on the parent